editor/store/effects.js

```diff
@@ -362,15 +366,57 @@ export default {
 					id,
 				} );
 				dispatch( createSuccessNotice(
-					__( 'Block updated.' ),
-					{ id: SAVE_REUSABLE_BLOCK_NOTICE_ID }
+					isTemporary ? __( 'Block created.' ) : __( 'Block updated.' ),
+					{ id: REUSABLE_BLOCK_NOTICE_ID }
 				) );
 			},
 			( error ) => {
 				dispatch( { type: 'SAVE_REUSABLE_BLOCK_FAILURE', id } );
 				dispatch( createErrorNotice(
-					get( error.responseJSON, 'message', __( 'An unknown error occured.' ) ),
-					{ id: SAVE_REUSABLE_BLOCK_NOTICE_ID }
+					get( error.responseJSON, 'message', __( 'An unknown error occurred.' ) ),
+					{ id: REUSABLE_BLOCK_NOTICE_ID }
+				) );
+			}
+		);
+	},
+	DELETE_REUSABLE_BLOCK( action, store ) {
+		const { id } = action;
+		const { getState, dispatch } = store;
+
+		const allBlocks = getBlocks( getState() );
+		const associatedBlocks = allBlocks.filter( block => isReusableBlock( block ) && block.attributes.ref === id );
+		const associatedBlockUids = associatedBlocks.map( block => block.uid );
```

Comment:
Might a reducer here (i.e. `allBlocks.reduce...`) help avoid passing through the blocks list twice? Or is that a heavier operation? Just a tiny thought on optimisation I guess :-)

Reply:
You're absolutely right that using `reduce` would likely be faster, but I prefer to optimise for readability over performance unless it's an operation that's performance sensitive. Deleting a reusable block is a fairly uncommon operation, so I'm not very worried.